tests/python/unittest/test_numpy_op.py

```diff
@@ -3542,6 +3542,53 @@ def test_np_true_divide():
         assert_almost_equal(out_mx.asnumpy(), out_np, rtol=1e-3, atol=1e-3, use_broadcast=False)
 
 
+@with_seed()
+@use_np
+def test_np_append():
+    class TestAppend(HybridBlock):
+        def __init__(self, axis=None):
+            super(TestAppend, self).__init__()
+            self._axis = axis
+
+        def hybrid_forward(self, F, a, b):
+            return F.np.append(a, b, axis=self._axis)
+
+    def get_new_shape(shape, axis):
+        shape_lst = list(shape)
+        if axis is not None:
+            shape_lst[axis] = random.randint(0, 3)
+        return tuple(shape_lst)
+
+    for shape in [(0, 0), (2, 3)]:
+        for hybridize in [True, False]:
+            for axis in [0, 1, None]:
+                #test gluon
+                test_append = TestAppend(axis=axis)
+                if hybridize:
+                    test_append.hybridize()
+
+                a = mx.nd.random.uniform(-1.0, 1.0, shape=get_new_shape(shape, axis)).as_np_ndarray()
+                a.attach_grad()
```

Comment:
`attach_grad` takes three kinds of `grad_req`s: `write`, `add`, and `null`. Please test these three cases respectively.

Reply:
ok